### PR TITLE
Complete automation of ARM template and removing user interactivity (added Login module)

### LINKED ISSUE #96: [Task] Complete automation of ARM template and removing user interactivity
Making sure the json file I use is a proper ARM template that can deploy the Azure function.

Fix user interactivity so that you can call the powershell script with parameters rather than having the user input after calling the script.

### PR DESCRIPTION
**Pull request by kevinles117**
Closes #96

Changes the script to an ARM template json and automates the deployment of the template.

### How to test

1. Pull my fork and branch in order to test the ARM template and script
`git remote set-url origin <url`
`git pull`
`git checkout <branch>`

2. Using the powershell terminal in VSCode you can call my script by using: 
`./script.ps1 -TenantID {} -SPApplicationID {} -SPSecret {} -SubscriptionID {} -resourceGroupName {} -location {} -templateUri ./Automation/FunctionAppUsersUserID/template.json`

Make sure to fill in the {} with the information from the secrets in nsc-kv-usw2-commonpsswds and to follow the correct naming conventions for the resourceGroupName

3. Wait for the script to finish and create everything successfully
4. Go to Azure and check if the resource group was created with the name you provided and there should be an App Service plan, Function App, Application Insights, and Storage Account that was created inside the resource group. 

After confirming everything was successfully created, delete the resource group!

### CODE REVIEW

**Review by AllisonBC — APPROVED** [review]
Test produced resources as described!

image

File: Automation/FunctionAppUsersUserID/script.ps1 [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    [string] 
+    $resourceGroupName,
+
+    [Parameter(Mandatory=$true)] 
+    [string]
+    $location,
+
+    [Parameter(Mandatory=$true)] 
+    [string]
+    $templateUri
```
Comment (theanticrumpet) [inline Automation/FunctionAppUsersUserID/script.ps1:28]: This should be "URL" not "URI"

**Review by theanticrumpet — APPROVED** [review]
I also got resources created, though it looks different than Allison's
test kevin

The resource naming convention for the resources doesn't make me happy, but it's functional. Also, for some reason when I would navigate my powershell to the folder containing your script and run it, it then wouldn't take `".\template.json"` as an acceptable template path, kept trying to look in C:\WINDOWS\system32\template.json instead of in the folder it was run in. I don't think that's on you but I have vague memories of someone else's not needing the full path.

**Comment by yergu** [comment]
I run your script it deployed successfully , all resources created 
image
I like how you modularized your script

**Review by yergu — APPROVED** [review]

File: Automation/FunctionAppUsersUserID/script.ps1 (hunk 1 above)
Comment (kevinles117) [reply Automation/FunctionAppUsersUserID/script.ps1:28]: Yeah I'll look into the names of the resources, and whoops thanks for the heads up! I'll have that changed!

**Pull request merged**